wicketd/src/update_tracker.rs

```diff
@@ -1844,6 +2001,151 @@ impl UpdateContext {
         })
     }
 
+    async fn interrogate_rot_bootloader(
+        &self,
+        rot_bootloader: &[ArtifactIdData],
+    ) -> Result<StepResult<Option<RotInterrogation>>, UpdateTerminalError> {
+        // We have a known set of bootloader FWID that don't have cabooses.
+        static KNOWN_MISSING_CABOOSE: [&str; 18] = [
+            "1122095f4a3797db8a7d6279ae889ddde0316631f1f3bc204bdc39c2d75707af",
+            "1525832a663024f6421c13c0f7c7d9e9b32ebf433898565a2ad8112e7d237ead",
+            "29fc0d31e1739865c7f3d4bb5f5b86779db92a65a2decbd59e42f6e95dd84698",
+            "37aa40d0ea12e1290477a84014cd03dbc6fa9817223d1546a10847510d75c383",
+            "53cb91f4a3fbb69efa733a9eb326bd9f71c849782b0eea4306ebc66620158d44",
+            "60effb7fd6c4780138887e0d65c9e9b9c8447ce4ea3ea71e08194aec2847b185",
+            "77b8fc4308221dfe123d93431c21b57fa896db65c015ca82e22a337c7aa7cd77",
+            "77c2b94e3a83fc6b3c8924d38b0d23ac7c1e7a15defa910ee3f850b41af9ca4c",
+            "8c58b2272fe2da219ab0757ff27398b8d4a459eb4e75c32c782f98d684269352",
+            "9dd79a4e7609bd4af8e39a03f77b997b35f5050409a2ecd19de1e7d16184b1f3",
+            "b123a0f683f4e7b60238840139c9f3dbfe2b2c61597d9cdd4e92c718f7f98bb7",
+            "ba08df44e7282a1daeae2d9346b99ca741bfc2649c12aa8292f413a1c84d80b7",
+            "bfa9adfc127886aeaa1ac58d30c07c76e89592c29fc83dfa88062e7f3a48335e",
+            "c23a53858e94932a95945f28730e41ae4a2d1a8db4776283245eda143b6b2994",
+            "e7ec5dae7ac462cc7f7561a91ef244a2ece0894ff212995fcccb1e86438cb665",
+            "ee688a237a480e9fd111a7f70cc4c6f9ac837dcac65a01e7cfa29f7c28545d07",
+            "f31442015da37523a13ffaa173b4dfe0b069c6d890cf1c9748a898001fe4110e",
+            "fa73f26fb73b27b5db8f425320e206df5ebf3e137475d40be76b540ea8bd2af9",
+        ];
+
+        // We already validated at repo-upload time there is at least one RoT
+        // artifact available and that all available RoT artifacts are the same
+        // version, so we can unwrap the first artifact here and assume its
+        // version matches any subsequent artifacts.
+        // TODO this needs to be fixed for multi version to work!
```

Comment:
I assume this TODO is for a subsequent PR? Will that be similar to the changes to allow multi-version non-bootloader RoT images?

Reply:
Yes, those changes should be very similar.